src/apps/vpn/externalophandler.cpp

```diff
@@ -63,10 +65,14 @@ bool ExternalOpHandler::request(Op op) {
       vpn->requestAbout();
       break;
     case OpActivate:
-      vpn->controller()->activate();
+      TaskScheduler::deleteTasks();
+      TaskScheduler::scheduleTask(
+          new TaskControllerAction(TaskControllerAction::eActivate));
       break;
     case OpDeactivate:
-      vpn->controller()->deactivate();
+      TaskScheduler::deleteTasks();
+      TaskScheduler::scheduleTask(
+          new TaskControllerAction(TaskControllerAction::eDeactivate));
```

Comment:
Most of your changes are pretty clear but wondering about this, is this change related to keeping a copy of the server data?

Reply:
this is kind of unrelated. I just found that we are calling deactivate() directly instead of using tasks. This is bad because that could break the order of tasks.